# LYCEUM OF THE PHILIPPINES UNIVERSITY – LAGUNA
### College of Business and Accountancy

---

# BDAT05 — Preliminary Assessment
## Part 2A · Practical Demonstration — **Notebook A**

**Fundamentals of Predictive Analytics** · Term 1, AY 2026–2027 · Saturday, 12 September 2026

| | |
|---|---|
| **Duration** | **60 minutes** |
| **Dataset** | *Lavandera Ko* — a laundry chain you have never seen |
| **Conditions** | Open notes — including your own P2/P3 notebooks and code. **No AI of any kind.** |
| **After this** | Part 2B: you present THIS notebook to me, and defend it |

### The rules of the hour

1. **Five sections, in order.** Each ends with an **EVIDENCE** cell (print the number) and a **READING** sentence (what the number means). A number without a sentence is half an answer.
2. **Reuse your own code freely.** That is not cheating; that is why you built it.
3. **Nothing changes silently.** Every row you remove or repair gets counted, in print.
4. Stuck 10 minutes? Write what you *would* do in the reading cell and move on — method earns marks even when code fights back.

---

*Course Facilitator: Jerald B. Bongalos*


## Setup — run once, at the start

In [1]:
STUDENT_NAME = "ALCANTARA_ANGUS_JULLIAN_GOMEZ"        # <-- your full name
TRACK = "P6"               # <-- P4, P5, P6 or P7

# ---------------------------------------------------------------- do not edit below
from datetime import datetime
START_TIME = datetime.now()
NOTEBOOK_CODE = "A"
print(f"Notebook {NOTEBOOK_CODE} — {STUDENT_NAME or 'NAME NOT ENTERED'} ({TRACK or '?'})")
print(f"Started at {START_TIME:%H:%M} — you have 60 minutes")


Notebook A — ALCANTARA_ANGUS_JULLIAN_GOMEZ (P6)
Started at 07:47 — you have 60 minutes


## Load the data

Paste the three FILE_IDs from the board, then run.


In [2]:
import pandas as pd

def drive_url(fid):
    return f"https://drive.google.com/uc?export=download&id={fid}"

ORDERS_ID   = "1M7qcdCwZ86sZbXKv0OJLl6M26ZB2EktP"   # <-- lavandera_orders.csv
BRANCHES_ID = "1QD8yUR_ZkXtwSbZm0scKFiX7Xu7_I6sY"   # <-- lavandera_branches.csv
CALENDAR_ID = "1P4CQPEv2-5f401i-nxerjssmd4bNSV75"   # <-- lavandera_calendar.csv

orders   = pd.read_csv(drive_url(ORDERS_ID))
branches = pd.read_csv(drive_url(BRANCHES_ID))
calendar = pd.read_csv(drive_url(CALENDAR_ID))

print("orders:", orders.shape, "| branches:", branches.shape, "| calendar:", calendar.shape)


orders: (8000, 7) | branches: (6, 4) | calendar: (608, 7)


---

# §1 · The handshake  *(≈8 minutes)*

Run the full handshake on `orders`: `info()`, `head()`, missing counts. Then answer in the READING cell: **row count, column count, and every column whose dtype is wrong for what it holds** — with what each wrong type would silently break.


In [6]:
# §1 — your handshake code
# Step 1: Inspect the structure and data types of orders
orders.info()
orders.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8000 entries, 0 to 7999
Data columns (total 7 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   OrderID        8000 non-null   object 
 1   Date           7980 non-null   object 
 2   BranchID       8000 non-null   object 
 3   ServiceType    8000 non-null   object 
 4   KiloWeight     8000 non-null   object 
 5   Amount         8000 non-null   float64
 6   PaymentMethod  8000 non-null   object 
dtypes: float64(1), object(6)
memory usage: 437.6+ KB


,OrderID,Date,BranchID,ServiceType,KiloWeight,Amount,PaymentMethod
0,O11092,2025-07-24,L01,Wash & Fold,12.9,462.68,GCash
1,O15755,2026-01-22,L02,Dry Clean,3.6,450.11,Cash
2,O11565,2026-03-09,L04,Ironing,6.6,284.15,GCash
3,O14663,2025-12-18,L02,Wash & Fold,6.5,239.14,GCash
4,O12111,2026-06-08,L03,Beddings,8.4,558.21,Cash


In [8]:
# §1 EVIDENCE — fill in and run
S1_ROWS = 8000          # total rows
S1_WRONG_DTYPES = "KiloWeight int64"  # e.g. "ColA (object, should be numeric); ColB (...)"
print("Rows:", S1_ROWS, "| Wrong dtypes:", S1_WRONG_DTYPES)


Rows: 8000 | Wrong dtypes: KiloWeight int64


**§1 READING** *(edit this cell — one or two sentences: what did the handshake warn you about?)*

> ... The row count for the orders is 8000 rows then the column count is 7 columns, so kiloweight should be int64.


---

# §2 · Cleaning — Repair the branch identifiers  *(≈12 minutes)*

The `BranchID` column contains formatting variants of real branches (case, hyphens, stray spaces). **Repair them — do not delete them.** Then show the distinct BranchID values before and after, and print **how many rows** your repair recovered into valid branch codes.


In [15]:
# §2 — your cleaning code

orders["BranchID"].dropna().unique()

# Step 1: Keep the original BranchID values for comparison
orders["BranchID_original"] = orders["BranchID"]

orders[["BranchID_original", "BranchID"]].head()

,BranchID_original,BranchID
0,L01,L01
1,L02,L02
2,L04,L04
3,L02,L02
4,L03,L03


In [16]:
# Step 2: Repair BranchID formatting variants
orders["BranchID"] = (
    orders["BranchID"]
    .str.strip()
    .str.upper()
    .str.replace("-", "", regex=False)
)

In [10]:
# §2 EVIDENCE — fill in and run
S2_EVIDENCE = "13"   # Distinct values before: 'L01', 'L02', 'L04', 'L03', 'L06', 'L05', 'L-02', 'L07', 'l01', 'l04', 'L05 ', 'L03 ', 'L-06'  ·  after: ______  ·  rows repaired: ______
print(S2_EVIDENCE)


13


**§2 READING** *(edit — the decision and its reason, as a cleaning-log line: issue → decision → why)*

> ... Issue: There are 13 distict branchID values that has the same values.The decision is to repair or format the branch id that has a formatting issues. Why because it the branch id only have 7 branches.


---

# §3 · The grain — Build the branch-month revenue grain  *(≈12 minutes)*

Using your repaired IDs: build a table where **one row = one branch in one month**, with total `Amount`. Exclude rows with no date, and say so. **Print the row count** of your grain table.


In [22]:
# §3 — your grain code
# Step 1: Convert date to datetime and count rows with no date
orders["Date"] = pd.to_datetime(orders["Date"], format="%Y-%m-%d")

missing_date_count = orders["Date"].isna().sum()

print("Rows excluded due to missing date:", missing_date_count)

Rows excluded due to missing date: 20


In [25]:
# Step 2: Create one row per branch per month
orders_valid = orders.dropna(subset=["Date"]).copy()

orders_valid["Month"] = orders_valid["Date"].dt.to_period("M")

branch_month = (
    orders_valid
    .groupby(["BranchID", "Month"], as_index=False)["Amount"]
    .sum()
)

print("Rows in branch-month grain table:", len(branch_month))

Rows in branch-month grain table: 137


In [27]:
branch_month.head()

,BranchID,Month,Amount
0,L01,2025-01,45307.64
1,L01,2025-02,27422.51
2,L01,2025-03,45377.68
3,L01,2025-04,37470.56
4,L01,2025-05,50566.97


In [28]:
# §3 EVIDENCE — fill in and run
S3_GRAIN_ROWS = 137   # Rows in my grain table: 137
print("Grain rows:", S3_GRAIN_ROWS)


Grain rows: 137


**§3 READING** *(edit — state the grain in row language, and whether the count is what the calendar and the business would predict)*

> ... Each row represent one branch for one calendar month with amount equal to the total amount generated by the branch during the month. The calender and branch id would predict the total avgrevenue of that one branch.


---

# §4 · Diagnostic — Diagnostic — the duplicate orders  *(≈14 minutes)*

Something in `OrderID` violates uniqueness. **Find it and count it**: how many surplus rows does the file carry, and what would they silently inflate if left in?


In [30]:
# §4 — your diagnostic code
# Step 1: Find order IDs that appear more than once
duplicate_orderids = orders.loc[
    orders["OrderID"].duplicated(keep=False),
    "OrderID"
].value_counts()

print(duplicate_orderids)

OrderID
O13529    2
O17411    2
O13025    2
O15781    2
O15472    2
O12292    2
O10026    2
O11455    2
O11462    2
O16153    2
O11963    2
O10094    2
O12043    2
O15574    2
O17772    2
O16087    2
O15079    2
O14968    2
O15486    2
O10481    2
O16460    2
O16194    2
O12905    2
O13014    2
O10238    2
O15495    2
O13166    2
O14758    2
O10283    2
O10702    2
Name: count, dtype: int64


In [31]:
# §4 EVIDENCE — fill in and run
S4_EVIDENCE = "30"   # Surplus duplicate rows: ______
print(S4_EVIDENCE)


30


**§4 READING** *(edit — what you found, and what it would have silently done to an analysis that missed it)*

> ...i found a 30 surplus duplicate rows, it would double count those order whereever the rows are aggregated so it would inflate it.


---

# §5 · Judgment — Judgment — the ₱ outlier  *(≈14 minutes)*

Compute the IQR fences on `Amount`. Report how many rows sit beyond the upper fence — then look at your **single most extreme row** (its service, its weight, its amount) and RULE on it: **KEEP / FLAG / DROP**, with the reason a manager would accept. Numbers first, verdict second.


In [33]:
# §5 — computation (if your task needs it)
# Step 1: Compute the IQR and upper fence for amount
q1 = orders["Amount"].quantile(0.25)
q3 = orders["Amount"].quantile(0.75)
iqr = q3 - q1
upper_fence = q3 + (1.5 * iqr)

upper_outliers = orders[orders["Amount"] > upper_fence]

print("Q1:", q1)
print("Q3:", q3)
print("IQR:", iqr)
print("Upper fence:", upper_fence)
print("Rows beyond upper fence:", len(upper_outliers))

Q1: 270.5475
Q3: 546.6725
IQR: 276.125
Upper fence: 960.86
Rows beyond upper fence: 671


**§5 VERDICT** *(edit this cell — your numbers, then your ruling and its defense. This is the section you will be asked about first in Part 2B.)*

> ...Q1: 270.5475
Q3: 546.6725
IQR: 276.125
Upper fence: 960.86
Rows beyond upper fence: 671
Flag not drop, 671 transactions exceed the statistical upper fence of ₱960.86. Because this is a substantial group rather than an isolated anomaly, we should treat these as high-value transactions requiring review rather than automatically deleting legitimate business activity.


## Finish — run before you save

In [32]:
END_TIME = datetime.now()
print("=" * 52)
print(f"BDAT05 PRELIM — PART 2A · NOTEBOOK {NOTEBOOK_CODE}")
print("=" * 52)
print(f"Name     : {STUDENT_NAME}")
print(f"Track    : {TRACK}")
print(f"Duration : {(END_TIME - START_TIME).seconds // 60} minutes of 60")
print("-" * 52)
print("Save as: BDAT05_Prelim_Part2A_" + NOTEBOOK_CODE + "_" +
      (STUDENT_NAME.split()[-1].upper() if STUDENT_NAME else "SURNAME") + ".ipynb")
print("Upload to your prelim-assessment folder. Then we talk.")


BDAT05 PRELIM — PART 2A · NOTEBOOK A
Name     : ALCANTARA_ANGUS_JULLIAN_GOMEZ
Track    : P6
Duration : 52 minutes of 60
----------------------------------------------------
Save as: BDAT05_Prelim_Part2A_A_ALCANTARA_ANGUS_JULLIAN_GOMEZ.ipynb
Upload to your prelim-assessment folder. Then we talk.


https://chatgpt.com/share/6aa50fb6-5aec-83ec-8128-b4a79028a826